src/front/wgsl/mod.rs

```diff
@@ -1373,30 +1361,56 @@ impl Parser {
         let mut binding = None;
         // Perspective is the default qualifier.
         let mut interpolation = None;
+        let mut stage = None;
```

Comment:
Maybe our code base isn't there yet with syntax validation, but what happens if an user forgets to add the `stage` attribute? I think it will completely go through the front-end -> IR, and crashes at the back-end?

I see at line [1509](https://github.com/gfx-rs/naga/pull/200/files#diff-37ba88792a802ec39755d9da39194f86R1509) that it then gets interpreted as a normal function.

Reply:
If there is no `stage`, then it's not an entry point. IR will be built just fine. If we are talking about wgpu-core as the client, it will try to get the entry point information from IR, and it will not find the entry point, so this is where an error will be produced.